test/controllers/samples_controller_test.rb

```diff
@@ -349,4 +349,118 @@ class SamplesControllerTest < ActionDispatch::IntegrationTest
       }
     end
   end
+
+  test 'report_info should return cached copy on second request' do
+    post user_session_path, params: @user_params
+
+    url = report_info_url
+    get url
+
+    assert_response :success
+
+    first_runtime = @response.headers["X-Runtime"]
+    cache_header = @response.headers["X-IDseq-Cache"]
+    assert_equal "missed", cache_header
+
+    get url
+    assert_response :success
+
+    second_runtime = @response.headers["X-Runtime"]
+    cache_header = @response.headers["X-IDseq-Cache"]
+    assert_equal "requested", cache_header
+
+    # Second request should always be faster than the first.
+    # Typically it will be 10x faster.
+    assert first_runtime.to_f > second_runtime.to_f
```

Comment:
I think asserting for time spent is not a good way to check if the report has been properly cached. That assertion is brittle and might lead to random failures between runs.

I suggest assertion would only check for the response header `["X-IDseq-Cache"]` AND expected response content. The way this assertion is currently done, if the second run returned an empty report, this test would give a false positive, hurting the confidence level on our tests.

Reply:
Fair point about the runtime. I added that initially for my interest, not for reliability. 

I will add the assertion about the content. That should have been the first thing!